test/SparqlExpressionTest.cpp

```diff
@@ -204,3 +225,56 @@ TEST(SparqlExpression, PlusAndMinus) {
   testDivide(bByD, b, d);
   testDivide(dByB, d, b);
 }
+
+// Helper lambda to enable testing a unary expression in one line (see below).
+//
+// TODO: The tests above could also be simplified (and made much more readable)
+// in this vein.
+template <typename UnaryExpression, typename OperandType, typename OutputType>
+auto testUnaryExpression =
+    [](std::vector<OperandType>&& operand, std::vector<OutputType>&& expected) {
```

Comment:
The `&&` do not really harm here, but are currently (and since this is a cheap test) not really necessary. Are they an artifact of playing around with the constructor of `VectorWithMemoryLimit`?